src/ClientApp/src/components/pages/User.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from "react";
+import Box from "@mui/material/Box";
+import Toolbar from "@mui/material/Toolbar";
+import Container from "@mui/material/Container";
+import Paper from "@mui/material/Paper";
+import Dialog from "@mui/material/Dialog";
+import UserList from "../UserList";
+import UserInputForm from "../UserInputForm";
+import ConfirmationDialog from "../ConfirmationDialog";
+
+export function User() {
+  const [users, setUsers] = useState([]);
+  const [openForm, setOpenForm] = useState(false);
+  const [openConfirmation, setOpenConfirmation] = useState(false);
+  const [currentUser, setCurrentUser] = useState(false);
+
+  const onDelete = (user) => {
+    setOpenConfirmation(true);
+    setCurrentUser(user);
+  };
+
+  const confirm = (confirmation) => {
+    if (confirmation) {
+      deleteUser(currentUser);
+    }
+    setOpenConfirmation(false);
+  };
+
+  const openEditForm = (user) => {
+    setCurrentUser(user);
+    setOpenForm(true);
+  };
+
+  // Edit user with the specified id.
+  async function editUser(data) {
+    const updatedUser = currentUser;
+    Object.entries(data).forEach(([key, value]) => {
+      updatedUser[key] = value;
+    });
+    const response = await fetch("/user", {
+      method: "PUT",
+      cache: "no-cache",
+      credentials: "same-origin",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(updatedUser, (key, value) => {
+        if (key === "role") {
+          return parseInt(value);
+        }
+        return value;
+      }),
+    });
+    if (response.ok) {
+      const updatedUsers = users.map((s) => (s.id === updatedUser.id ? updatedUser : s));
+      setUsers(updatedUsers);
+    }
+  }
+
+  // Delete user with the specified id.
+  async function deleteUser(user) {
+    const response = await fetch("/user?id=" + user.id, {
+      method: "DELETE",
+    });
+    if (response.ok) {
+      const updatedUsers = users.filter((s) => s.id !== currentUser.id);
+      setUsers(updatedUsers);
+    }
+  }
+
+  // Get all users from database.
+  useEffect(() => {
+    let fetchurl = "/user";
+    fetch(fetchurl)
+      .then((response) => response.json())
+      .then((users) => {
+        setUsers(users);
+      });
+  }, []);
+
+  return (
+    <Box
+      component="main"
+      sx={{
+        backgroundColor: (theme) =>
+          theme.palette.mode === "light" ? theme.palette.grey[100] : theme.palette.grey[900],
```

Comment:
PP: Das können wir glaube  ich in der ganzen Applikation weglassen. Ich hatte das aus einer Standardkomponente übernommen, aber da wir zum jetzigen Zeitpunkt ohnehin keine darkmode unterstützen werden können wir das streichen.